src/NzbDrone.Core/MediaFiles/EpisodeFileMovingService.cs

```diff
@@ -113,10 +126,35 @@ private EpisodeFile TransferFile(EpisodeFile episodeFile, Series series, List<Ep
                 throw new SameFilenameException("File not moved, source and destination are the same", episodeFilePath);
             }
 
-            _diskTransferService.TransferFile(episodeFilePath, destinationFilePath, mode);
+            bool transfer = true;
 
             episodeFile.RelativePath = series.Path.GetRelativePath(destinationFilePath);
 
+            if (localEpisode is not null)
+            {
+                var scriptImportDecision = _scriptImportDecider.TryImport(episodeFilePath, destinationFilePath, localEpisode, mode);
+
+                switch (scriptImportDecision)
+                {
+                    case ScriptImportDecision.DeferMove:
+                        break;
+                    case ScriptImportDecision.RenameRequested:
+                        _updateMediaInfo.UpdateData(episodeFile, series);
+                        episodeFile.Path = null;
+                        MoveEpisodeFile(episodeFile, series, episodeFile.Episodes);
```

Comment:
Instead of passing all this logic back out of `ScriptImportDecider.Import`, let that handle getting the media info for the new file as well as getting the new file name and basically treats it as a `DeferMove` result.

`_buildFileNames.BuildFilePath(episodes, series, episodeFile, Path.GetExtension(episodeFile.RelativePath));` to calculate the new name and `_videoFileInfoReader.GetMediaInfo(path)` for the media info

Reply:
I don't quite follow. Deferring will trigger a call to the disk service to move, copy or hardlink the file. We don't know which of these it'll do. Here we want the move to _always_ be a move. And the paths for how it's moved around also are calculated before this? 

The current process for RenameRequested:
1. Script moves file to the path specified by Sonarr, essentially as though it was a regular move/copy/hardlink. 
2. Return code of the script first triggers Sonarr to recheck the MediaInfo, then the file will be renamed(i.e. moved) _within_ the library. The downloads folder doesn't get touched in this step at all.

A defer move doesn't move within the library, it still moves/copies/hardlinks from the downloads folder. I don't see how these are compatible.

Reply:
Ahh, it's already "imported" in this case, but this is still leaking more logic out than necessary, part of that is due to how `EpisodeFileMovingService` is structured (to actually get the new filename and move it), which might change if refactored. In the mean time it can be left as-is, but updating the mediainfo can be done within the script decider and this can just be responsible for taking care of the moving.

Reply:
Yeah, that's what I tried to do with the refactor before, but alright, I'll fix this and let's hope once this is done at least the actual code will be fine and I can move on to checking everything still tests ok and history cleanup.